### DISCUSSION TITLE
PETSc error

### DISCUSSION THREAD

**Discussion by chenyi-maker** [firedrakeproject/firedrake#2589] [Firedrake support]
Hello everyone
    I run the Firedrake code in parallel on the server. My server configuration is included in Attach File, it has enough for more than 15 cores,  but I can only call up to 15 cores for parallelism. If I call more than 15 cores, he will always report a PETSc error: Caught signal number 7 BUS: Bus error , possibly illegal memory access.  My Firedrake is integrated using the Docker environment.

Thank you for your answers
![屏幕截图 2022-10-17 103356](https://user-images.githubusercontent.com/115963126/196077049-8e435995-f55c-4054-a4a0-66b0a5b5ccb0.png)
![屏幕截图 2022-10-17 103500](https://user-images.githubusercontent.com/115963126/196077052-6145c057-d750-4adf-b11b-cb16f7f1febe.png)

**Comment by danshapero** [comment]
I think what you're running into relates to the what communication mechanism, or *device*, MPICH gets built with. The default is to build MPICH using the "nemesis" device. Nemesis enables certain shared-memory optimizations, with the idea that it can work much faster on a workstation or a single machine with very little tweaking on the user's part. But the nemesis device has a limitation that you can't oversubscribe it -- you can use only as many MPI processes as you have cores.

You can change how MPICH gets built so that it uses a different communication device when you build Firdrake; see the [Firedrake documentation](https://firedrakeproject.org/parallelism.html#installing-for-parallel-use). There's some good information about the different communication devices in section 4 of the [MPICH documentation](https://www.mpich.org/static/downloads/3.3/mpich-3.3-README.txt).

**Reply by chenyi-maker** [reply]
Hello, thank you very much for your answer. What I want to consult is that my server device now has 28*4=112 cores, but I have only invoked 15 cores, which is within the limit. Why does PETSC error still occur?
![屏幕截图 2022-10-17 103500](https://user-images.githubusercontent.com/115963126/196373306-8a231c5b-d8ab-44d1-a4b1-44ee9880ffee.png)

**Reply by danshapero** [reply]
Sorry, I should have noticed that! I take it back, I have no idea what's going wrong then.

**Reply by chenyi-maker** [reply]
It doesn't matter I still appreciate your advice on how to solve the problem.
